<a href="https://colab.research.google.com/github/zaidbinnaveed/FlyRank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [ ]:
!git clone https://github.com/zaidbinnaveed/FlyRank-ML-Internship.git
%cd FlyRank-ML-Internship

fatal: destination path 'FlyRank-ML-Internship' already exists and is not an empty directory.
/content/FlyRank-ML-Internship


## 1. My lane (or freestyle) and why

I am selecting Lane 2: Refresh / Content Opportunity Scoring. This lane is supported by strong preliminary evidence from the starter pipeline. A random forest model achieved 74% precision on its top 50 recommendations, compared to 24% for a rule-based baseline. It also maps to a concrete, actionable decision: prioritizing which pages a content team reviews first, making it well suited to a 7-week capstone.

In [ ]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# 1. Dataset size
print("Total pages in dataset:", len(df))

# 2. Pages currently trending downward
declining_count = (df["trend_direction"] == "down").sum()
print("Pages currently declining:", declining_count)

# 3. Pages that are outdated but still getting meaningful traffic
#    (strong candidates for review — evidence + inaction risk)
stale_visible_count = (
    (df["days_since_last_update"] >= 180) &
    (df["impressions_90d"] >= 500)
).sum()
print("Stale but visible pages:", stale_visible_count)

# 4. Baseline context: average traffic per page
avg_impressions = df["impressions_90d"].mean()
print("Average 90-day impressions per page:", round(avg_impressions, 1))


Total pages in dataset: 30000
Pages currently declining: 16262
Stale but visible pages: 17
Average 90-day impressions per page: 5200.4


## 2. The question: decision, action, cost of a wrong call

This work improves the decision of which pages a content team should review and update first, given limited time to review every page. A content editor or SEO specialist acts on this recommendation, deciding which flagged pages to rewrite, expand, or leave as is. A wrong recommendation carries a real cost. If a page is wrongly flagged as high priority, an editor wastes time on a page that did not actually need attention. If a genuinely declining page is missed instead, it keeps losing traffic or ranking unnoticed, and the problem grows harder to fix the longer it goes unaddressed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages in dataset:", len(df))

declining_count = (df["trend_direction"] == "down").sum()
print("Pages currently declining:", declining_count)

stale_visible_count = (
    (df["days_since_last_update"] >= 180) &
    (df["impressions_90d"] >= 500)
).sum()
print("Stale but visible pages:", stale_visible_count)


Total pages in dataset: 30000
Pages currently declining: 16262
Stale but visible pages: 17


## 4. Careful words: what I can and can't claim

I can claim that decline is common in this dataset, since 16262 out of 30000 pages are currently trending downward. I can also claim that staleness alone does not explain most declines, since only 17 pages are both stale and still visible. I cannot yet claim that any model will accurately predict which pages need review, or that fixing a page will actually cause it to recover, since no model has been built or tested.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.